pkg/kv/kvserver/rangefeed/scheduler_test.go

```diff
@@ -520,3 +532,86 @@ func TestNewSchedulerShards(t *testing.T) {
 		})
 	}
 }
+
+// TestSchedulerPriority tests that the scheduler correctly registers
+// and enqueues events for priority processors.
+func TestSchedulerPriority(t *testing.T) {
```

Comment:
We can reuse consumer that is used in other tests. We only need to either add priority to create consumer or have separate method to create with priority:

```
func TestProcessPriority(t *testing.T) {
	defer leaktest.AfterTest(t)()
	ctx := context.Background()
	stopper := stop.NewStopper()
	defer stopper.Stop(ctx)

	s := NewScheduler(SchedulerConfig{Workers: 1, PriorityWorkers: 1, ShardSize: 1})
	require.NoError(t, s.Start(ctx, stopper), "failed to start")
	c1 := createAndRegisterConsumerOrFail(t, s)
	c2 := createAndRegisterPriorityConsumerOrFail(t, s)
	c1.pause()
	s.Enqueue(c1.id, te1)
	c1.waitPaused()
	s.Enqueue(c2.id, te1)
	c2.requireHistory(t, time.Second*30, []processorEventType{te1})
	c1.resume()
	c1.requireHistory(t, time.Second*30, []processorEventType{te1})
	assertStopsWithinTimeout(t, s)
	c1.requireStopped(t, time.Second*30)
	c2.requireStopped(t, time.Second*30)
}
```
Basically copy-pasta of test above that verifies that one blocked consumer doesn't prevent the other to proceed.